library/utilities/src/main/kotlin/pl/mobite/lib/utilities/FragmentExtensions.kt

```diff
@@ -1,17 +1,17 @@
 package pl.mobite.lib.utilities
 
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.flowWithLifecycle
 import androidx.lifecycle.lifecycleScope
-import androidx.lifecycle.repeatOnLifecycle
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.FlowCollector
 import kotlinx.coroutines.launch
 
-fun <T>Fragment.collectFlowWhenStarted(flow: Flow<T>, flowCollector: FlowCollector<T>) {
-    viewLifecycleOwner.lifecycleScope.launch {
-        viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
-            flow.collect(flowCollector)
-        }
+
+fun <T>Flow<T>.collectWithViewLifecycle(fragment: Fragment, flowCollector: FlowCollector<T>) {
```

Comment:
While I agree that it's better to have this function to be extension of `Flow` rather than `Fragment`, the function itself is doing too much with `viewLifecycleOwner`. I would keep this method more simple and make it suspend
```
suspend fun <T> Flow<T>.collectWithLifecycle(lifecycle: Lifecycle, collector: FlowCollector<T>) {
    flowWithLifecycle(lifecycle).collect(collector)
}
```

while leaving privilage of creating coroutines for scope owner itself:
```
viewLifecycleOwner.lifecycleScope.launch {
            viewModel.viewState.collectWithLifecycle(viewLifecycleOwner.lifecycle, ::render)
}
```

Reply:
In other words (from my experience working with coroutines), I see much more advantages of using suspend functions and call them from coroutines, created from most top, rather than moving coroutine scope to bottom and allow creation of coroutines there.

Reply:
I agree [dad4c56](https://github.com/mdabrowski89/TodoApp/pull/5/commits/dad4c566bc6d52638aa3a93e9877cc9610ab2e67)